cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/JsonOutputFormatter.cs

```diff
@@ -0,0 +1,82 @@
+using System.CommandLine;
+using System.Text.Json;
+using Spectre.Console;
+
+namespace Microsoft.Kiota.Cli.Commons.IO;
+
+/// <summary>
+/// The JSON output formatter
+/// </summary>
+public class JsonOutputFormatter : IOutputFormatter
+{
+    private readonly IAnsiConsole _ansiConsole;
+
+    /// <summary>
+    /// Creates a new JSON output formatter with a default console
+    /// </summary>
+    public JsonOutputFormatter() : this(AnsiConsole.Console)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new JSON output formatter with the provided console
+    /// </summary>
+    /// <param name="console">The console to use</param>
+    public JsonOutputFormatter(IAnsiConsole console)
+    {
+        this._ansiConsole = console;
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions options, CancellationToken cancellationToken = default)
+    {
+        string resultStr;
+
+        if (options is IJsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            using var result = await ProcessJsonAsync(content, jsonOptions.OutputIndented, cancellationToken);
+            using var r = new StreamReader(result);
+            resultStr = await r.ReadToEndAsync();
```

Comment:
```suggestion
            resultStr = await r.ReadToEndAsync().ContinueWith(x => Task.FromResult(x), cancellationToken);
```

what about this instead then?

Reply:
It still won't make the `ReadToEndAsync` function support cancellation. I think the `cancellationToken` here will cancel the task that returns from the action continuation. It seems unnecessary.

Reply:
correct, disregard